run.py

```diff
@@ -197,13 +201,160 @@ def get_files(args, dirname, min_size=1, min_date=''):
    return set(ls_s3_dir("%s/%s/%s/" % (S3_BUCKET, args.bioproject, dirname),
                         min_size=min_size, min_date=min_date))
 
+def ribocounts(args, subset_size=1000):
+    """Fast algorithm to compute fraction of reads identified as rRNA by RiboDetector"""
+
+    available_inputs = get_files(args, "cleaned",
+                                 # tiny files are empty; ignore them
+                                 min_size=100)
+    existing_outputs = get_files(args, "ribocounts", min_date='2023-10-10')
+
+    def first_subset_fastq(file_paths, subset_size):
+        """
+        Selects the first subset of reads from gzipped fastq files.
+        Assumes that paired-end reads are in the same order in both files.
+        """
+        print(f"Counting reads in input and selecting the first {subset_size}...")
+        output_files = []
+        total_reads = 0
+        # Count the total number of reads only for the first input file
+        with gzip.open(file_paths[0], 'rt') as f:
+            total_reads = sum(1 for _ in FastqGeneralIterator(f))
+
+        for fp in file_paths:
+            actual_subset_size = 0
```

Comment:
comment what this is for

(I'm guessing it's because some files have fewer than `subset` reads, but it would be good to make that clear)

Reply:
Updated